process/smartContract/processorV2/failureContext.go

```diff
@@ -0,0 +1,58 @@
+package processorV2
+
+import vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+
+type failureContext struct {
+	sc            *scProcessor
+	processFail   bool
+	txHash        []byte
+	errorMessage  string
+	returnMessage []byte
+	snapshot      int
+	gasLocked     uint64
+	logs          []*vmcommon.LogEntry
+}
+
+func NewFailureContext(sc *scProcessor) *failureContext {
```

Comment:
providing the sc processor pointer here suggest that the failureContext and scProcessor are closely tight together.
In this case, I suggest making the scProcessorV2 keep a pointer of type failure context inside it. In this way you will not need to pass it around on all methods. You will need to reset it whenever the sc processor needs to start processing something.

Reply:
In one sense you are right, however we do not have any accumulator in this sense at SCProcessorLevel - I think it is better to keep it that way. dependency on scProcessor can be taken out if instead on makeSnapshot - we have setSnapshot(sc.accounts.journallen)

Reply:
done